apps/emqx/src/emqx_corrupt_namespace_config_checker.erl

```diff
@@ -0,0 +1,118 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2025 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%--------------------------------------------------------------------
+-module(emqx_corrupt_namespace_config_checker).
+
+-include_lib("emqx/include/logger.hrl").
+-include_lib("snabbkaffe/include/trace.hrl").
+
+%% API
+-export([
+    start_link/0,
+
+    check/1
+]).
+
+%% `gen_server' API
+-export([
+    init/1,
+    handle_continue/2,
+    handle_call/3,
+    handle_cast/2,
+    handle_info/2
+]).
+
+%%------------------------------------------------------------------------------
+%% Type declarations
+%%------------------------------------------------------------------------------
+
+%% Calls/casts/infos
+-record(check_all, {}).
+-record(check, {ns}).
+
+%%------------------------------------------------------------------------------
+%% API
+%%------------------------------------------------------------------------------
+
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+check(Namespace) ->
+    gen_server:cast(?MODULE, #check{ns = Namespace}).
+
+%%------------------------------------------------------------------------------
+%% `gen_server' API
+%%------------------------------------------------------------------------------
+
+init(_) ->
+    case mria_config:whoami() of
+        replicant ->
+            ignore;
+        _ ->
+            State = #{},
+            {ok, State, {continue, #check_all{}}}
+    end.
+
+handle_continue(#check_all{}, State) ->
+    NamespaceErrors = emqx_config:get_all_namespace_config_errors(),
+    raise_alarms(NamespaceErrors),
+    ?tp("corrupt_ns_checker_started", #{}),
+    {noreply, State}.
+
+handle_call(Call, _From, State) ->
+    {reply, {error, {unknown_call, Call}}, State}.
+
+handle_cast(#check{ns = Namespace}, State) ->
+    do_check(Namespace),
+    ?tp("corrupt_ns_checker_checked", #{ns => Namespace}),
+    {noreply, State};
+handle_cast(_Cast, State) ->
+    {noreply, State}.
+
+handle_info(_Info, State) ->
+    {noreply, State}.
+
+%%------------------------------------------------------------------------------
+%% Internal fns
+%%------------------------------------------------------------------------------
+
+raise_alarms(NamespaceErrors) when is_list(NamespaceErrors) ->
+    lists:foreach(fun raise_alarms/1, NamespaceErrors);
+raise_alarms({Namespace, HoconErrorsPerRoot0}) ->
+    HoconErrorsPerRoot = format_errors(HoconErrorsPerRoot0),
+    emqx_alarm:safe_activate(
```

Comment:
If the number of Namespaces is huge, won't be EMQX or Dashboard overflooded with alarma?

Reply:
Yes, I think so...  
I'm not sure how we could handle such a case.  :crying_cat_face:

Reply:
Maybe we should raise one fat alarm with all namespaces?  :thinking:

Reply:
Changed to raise a single alarm for all namespaces.

:checkered_flag: